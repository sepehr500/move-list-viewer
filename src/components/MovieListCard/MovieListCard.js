import React from "react";
import PropTypes from "prop-types";
import Rater from "react-rater";

const MovieListCard = ({
  title,
  thumbnailPath,
  releaseDate,
  score,
  onClick
}) => (
  <div className="shadow-1 b--black-40 bg-white mr3 flex flex-column br3 mv3 ba bw1 dark-gray b--black">
    <div className="dib center">
      <img alt={`${title} poster`} src={thumbnailPath} />
    </div>
    <div style={{ height: "4rem" }} className="tc">
      <div className="pt3 ph2" style={{ maxWidth: "200px" }}>
        <b>{title}</b>
      </div>
    </div>
    <div className="ph2 mv2">
      <div className="mb1">Release date: {releaseDate}</div>
    </div>
    <div style={{ fontSize: "40px" }} className="f1">
      <Rater interactive={false} total={5} rating={score} />
    </div>
    <button
      onClick={onClick}
      className="pb3 f3 w-100 br2 Details-button"
      value="details"
    >
      details
    </button>
  </div>
);

MovieListCard.propTypes = {
  title: PropTypes.string.isRequired,
  thumbnailPath: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired
};

export { MovieListCard };
