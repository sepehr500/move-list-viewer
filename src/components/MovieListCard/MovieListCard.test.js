import React from "react";
import renderer from "react-test-renderer";
import { MovieListCard } from "./MovieListCard";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MovieListCard
        key={1}
        title="Gone with the wind"
        thumbnailPath="/asdf9n.jpg"
        releaseDate="2016-08-04"
        score={4}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
