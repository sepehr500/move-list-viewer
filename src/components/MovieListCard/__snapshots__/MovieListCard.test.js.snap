// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MovieListCard
  onClick={[MockFunction]}
  releaseDate="2016-08-04"
  score={4}
  thumbnailPath="/asdf9n.jpg"
  title="Gone with the wind"
>
  <div
    className="shadow-1 b--black-40 bg-white mr3 flex flex-column br3 mv3 ba bw1 dark-gray b--black"
  >
    <div
      className="dib center"
    >
      <img
        alt="Gone with the wind poster"
        src="/asdf9n.jpg"
      />
    </div>
    <div
      className="tc"
      style={
        Object {
          "height": "4rem",
        }
      }
    >
      <div
        className="pt3 ph2"
        style={
          Object {
            "maxWidth": "200px",
          }
        }
      >
        <b>
          Gone with the wind
        </b>
      </div>
    </div>
    <div
      className="ph2 mv2"
    >
      <div
        className="mb1"
      >
        Release date: 
        2016-08-04
      </div>
    </div>
    <div
      className=" center f1"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      <Rater
        interactive={false}
        rating={4}
        total={5}
      >
        <div
          className="react-rater"
        >
          <div
            key="star-0"
            onClick={null}
            onMouseOver={null}
          >
            <Star
              isActive={true}
              isActiveHalf={false}
              isDisabled={true}
              willBeActive={false}
            >
              <div
                className="react-rater-star is-disabled is-active"
              >
                ★
              </div>
            </Star>
          </div>
          <div
            key="star-1"
            onClick={null}
            onMouseOver={null}
          >
            <Star
              isActive={true}
              isActiveHalf={false}
              isDisabled={true}
              willBeActive={false}
            >
              <div
                className="react-rater-star is-disabled is-active"
              >
                ★
              </div>
            </Star>
          </div>
          <div
            key="star-2"
            onClick={null}
            onMouseOver={null}
          >
            <Star
              isActive={true}
              isActiveHalf={false}
              isDisabled={true}
              willBeActive={false}
            >
              <div
                className="react-rater-star is-disabled is-active"
              >
                ★
              </div>
            </Star>
          </div>
          <div
            key="star-3"
            onClick={null}
            onMouseOver={null}
          >
            <Star
              isActive={true}
              isActiveHalf={false}
              isDisabled={true}
              willBeActive={false}
            >
              <div
                className="react-rater-star is-disabled is-active"
              >
                ★
              </div>
            </Star>
          </div>
          <div
            key="star-4"
            onClick={null}
            onMouseOver={null}
          >
            <Star
              isActive={false}
              isActiveHalf={false}
              isDisabled={true}
              willBeActive={false}
            >
              <div
                className="react-rater-star is-disabled"
              >
                ★
              </div>
            </Star>
          </div>
        </div>
      </Rater>
    </div>
    <button
      className="pb3 f3 w-100 br--bottom br2 Details-button"
      onClick={[MockFunction]}
      value="details"
    >
      details
    </button>
  </div>
</MovieListCard>
`;
